### PR TITLE
Connection setup

### PR DESCRIPTION
**Pull request by cr2david**
Can now register and login

token gets stored app scope so we can use it easily for the demo (when we get profile going)

todo:
- figure out proper way to store the token
- use okhttp to intercept all requests and just force auth header into them
- write general de-serializer because the data deserialization is gonna use up 100 files

### CODE REVIEW

File: android/app/src/main/java/com/chorestory/app/ParentLoginActivity.java [hunk 1]
```diff
@@ -40,18 +61,35 @@ public void onClick(View v) {
...
-                // TODO: if authentication unsuccessful, enable loginButton, display toast
-                // TODO: use Snackbar instead; move existing view up when Snackbar appears
-//                toast = Toast.makeText(ParentLoginActivity.this,
-//                        "Invalid username or password",
-//                        Toast.LENGTH_SHORT);
-//                toast.setGravity(Gravity.CENTER_HORIZONTAL, 0, 400);
-//                toast.show();
-//                logInButton.setEnabled(true);
+                            // TODO: Navigate to profile page
+                        } else{
```
Comment (jimzhaozhengwang) [inline android/app/src/main/java/com/chorestory/app/ParentLoginActivity.java]: in `Android Studio` 
hover over `Code` menu
click `Reformat Code` 
on all modified files please

File: android/app/src/main/java/com/chorestory/app/MainActivity.java [hunk 2]
```diff
@@ -8,8 +8,15 @@
...
 import java.util.Arrays;
 
+import javax.inject.Inject;
+
+import okhttp3.OkHttpClient;
+
 public class MainActivity extends ChoreStoryActivity {
 
+    @Inject
+    OkHttpClient okHttpClient;
```
Comment (jimzhaozhengwang) [inline android/app/src/main/java/com/chorestory/app/MainActivity.java]: this isn't being used

File: android/app/src/main/java/com/chorestory/app/ParentLoginActivity.java [hunk 3]
```diff
@@ -40,18 +61,35 @@ public void onClick(View v) {
...
                 password = passwordEditText.getText().toString();
 
-                // TODO: Authenticate user
+                LoginRequest loginRequest = new LoginRequest(username, password);
+                Call<LoginResponse> loginQuery = retrofitInterface.login(loginRequest);
+
+                loginQuery.enqueue(new Callback<LoginResponse>() {
+                    @Override
+                    public void onResponse(Call<LoginResponse> call, Response<LoginResponse> response) {
+                        if (response.isSuccessful()) {
```
Comment (jimzhaozhengwang) [inline android/app/src/main/java/com/chorestory/app/ParentLoginActivity.java]: does `response` need a null check first?

File: android/app/src/main/java/com/chorestory/app/ParentLoginActivity.java [hunk 4]
```diff
@@ -40,18 +61,35 @@ public void onClick(View v) {
...
+                                    "Invalid username or password",
+                                    Toast.LENGTH_LONG);
+                            toast.setGravity(Gravity.CENTER_HORIZONTAL, 0, 400);
+                            toast.show();
+                            logInButton.setEnabled(true);
+                        }
+                    }
+                    @Override
+                    public void onFailure(Call<LoginResponse> call, Throwable t) {
+                        toast.show();
```
Comment (jimzhaozhengwang) [inline android/app/src/main/java/com/chorestory/app/ParentLoginActivity.java]: need to make text and set gravity first
see line 79

File: android/app/src/main/java/com/chorestory/app/CreateClanActivity.java [hunk 5]
```diff
@@ -44,8 +63,35 @@ public void onClick(View v) {
...
 
-                // TODO: verification
-                //  if successful register new clan & user, else display snackbar/toast
+                RegisterRequest registerRequest = new RegisterRequest(clanName, username, password);
+                Call<LoginResponse> registerQuery = retrofitInterface.register(registerRequest);
+
+                registerQuery.enqueue(new Callback<LoginResponse>() {
+                    @Override
+                    public void onResponse(Call<LoginResponse> call, Response<LoginResponse> response) {
+                        if (response.isSuccessful()) {
```
Comment (jimzhaozhengwang) [inline android/app/src/main/java/com/chorestory/app/CreateClanActivity.java]: does `response` need a null check first?

File: android/app/src/main/java/com/chorestory/app/CreateClanActivity.java [hunk 6]
```diff
@@ -44,8 +63,35 @@ public void onClick(View v) {
...
+                                    "Something went wrong!",
+                                    Toast.LENGTH_LONG);
+                            toast.setGravity(Gravity.CENTER_HORIZONTAL, 0, 400);
+                            toast.show();
+                            signUpButton.setEnabled(true);
+                        }
+                    }
+                    @Override
+                    public void onFailure(Call<LoginResponse> call, Throwable t) {
+                        toast.show();
```
Comment (jimzhaozhengwang) [inline android/app/src/main/java/com/chorestory/app/CreateClanActivity.java]: need to make text and set gravity first
see line 81

Comment (keller00) [reply android/app/src/main/java/com/chorestory/app/CreateClanActivity.java]: would negative gravity make it fly upwards forever, if so can we do that?

File: android/app/src/main/java/com/chorestory/helpers/TokenHandler.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
+package com.chorestory.helpers;
+
+public class TokenHandler {
+
+    private String authToken = null;
+
+    // This will eventually be setup to pull from some storage system
+    // Undecided ActivityManager/keystore
+    public void setToken(String token) {
+        this.authToken = token + ":";
```
Comment (keller00) [inline android/app/src/main/java/com/chorestory/helpers/TokenHandler.java]: This is only right for parent account, not children

File: android/app/src/main/java/com/chorestory/app/ParentLoginActivity.java (hunk 3 above)
Comment (cr2david) [reply android/app/src/main/java/com/chorestory/app/ParentLoginActivity.java]: nope, response and response.body() are handled but I'll add a check for getToken()

File: android/app/src/main/java/com/chorestory/app/CreateClanActivity.java [hunk 8]
```diff
@@ -44,8 +63,40 @@ public void onClick(View v) {
...
+                                    Toast.LENGTH_LONG);
+                            toast.setGravity(Gravity.CENTER_HORIZONTAL, 0, 400);
+                            toast.show();
+                            signUpButton.setEnabled(true);
+                        }
+                    }
+
+                    @Override
+                    public void onFailure(Call<LoginResponse> call, Throwable t) {
+                        toast = Toast.makeText(CreateClanActivity.this,
```
Comment (jimzhaozhengwang) [inline android/app/src/main/java/com/chorestory/app/CreateClanActivity.java]: refactor this into a `showToast` method

File: android/app/src/main/java/com/chorestory/app/ParentLoginActivity.java [hunk 9]
```diff
@@ -40,18 +61,40 @@ public void onClick(View v) {
...
-                // TODO: use Snackbar instead; move existing view up when Snackbar appears
-//                toast = Toast.makeText(ParentLoginActivity.this,
-//                        "Invalid username or password",
-//                        Toast.LENGTH_SHORT);
-//                toast.setGravity(Gravity.CENTER_HORIZONTAL, 0, 400);
-//                toast.show();
-//                logInButton.setEnabled(true);
+                    @Override
+                    public void onFailure(Call<LoginResponse> call, Throwable t) {
+                        toast = Toast.makeText(ParentLoginActivity.this,
```
Comment (jimzhaozhengwang) [inline android/app/src/main/java/com/chorestory/app/ParentLoginActivity.java]: refactor this into a `showToast` method

**Pull request merged**